neuro_extras/main.py

```diff
@@ -474,19 +431,107 @@ async def _nonstorage_cp(
+        command = "tar"
+        args = ["zxvf", str(file), "-C", str(destination_url.path)]
+    elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
+        command = "tar"
+        args = ["jxvf", str(file), "-C", str(destination_url)]
+    elif suffixes[-1] == ".tar":
+        command = "tar"
+        args = ["xvf", str(file), "-C", str(destination_url)]
+    elif suffixes[-1] == ".gz":
+        command = "gunzip"
+        args = ["--keep", str(file), str(destination_url) + file.name[:-3]]
+    elif suffixes[-1] == ".zip":
+        command = "unzip"
+        args = [str(file), "-d", str(destination_url)]
+    else:
+        raise ValueError(
+            f"Don't know how to extract file {file.name}"
+            f"Supported archive types: {', '.join(SUPPORTED_ARCHIVE_TYPES)}"
+        )
+
+    click.echo(f"Running '{command} {' '.join(args)}'")
+    subprocess = await asyncio.create_subprocess_exec(command, *args)
+    returncode = await subprocess.wait()
+    if rm_src:
+        _rm_by_url(source_url)
+    if returncode != 0:
+        raise click.ClickException(f"Extraction failed: {subprocess.stderr}")
+
+
+async def _compress(source_url: URL, destination_url: URL, rm_src: bool) -> None:
+    file = Path(destination_url.path.split("/")[-1])
+    suffixes = file.suffixes
+    if suffixes[-2:] == [".tar", ".gz"] or suffixes[-1] == ".tgz":
+        command = "tar"
+        args = [
+            "zcf",
+            str(destination_url.path),
+            "-C",
+            str(Path(source_url.path).parent),
+            f"--exclude={str(destination_url.name)}",
+            ".",
+        ]
+    elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
+        command = "tar"
+        args = [
+            "jcf",
+            str(destination_url.path),
+            f"--exclude={str(destination_url.name)}",
+            str(source_url.path),
+        ]
+    elif suffixes[-1] == ".tar":
+        command = "tar"
+        args = [
+            "cf",
+            str(destination_url.path),
+            f"--exclude={str(destination_url.name)}",
+            str(source_url.path),
+        ]
+    elif suffixes[-1] == ".gz":
+        command = "gzip"
+        args = ["-r", str(destination_url.path), str(source_url.path)]
+    elif suffixes[-1] == ".zip":
+        command = "zip"
+        args = ["-r", str(destination_url.path), str(source_url.path)]
+    else:
+        raise ValueError(
+            f"Don't know how to compress to archive type {file.name}. "
+            f"Supported archive types: {', '.join(SUPPORTED_ARCHIVE_TYPES)}"
+        )
+
+    click.echo(f"Running '{command} {' '.join(args)}'")
+    subprocess = await asyncio.create_subprocess_exec(command, *args)
+    returncode = await subprocess.wait()
+    if rm_src:
+        _rm_by_url(source_url)
+    if returncode != 0:
+        raise click.ClickException(f"Compression failed: {subprocess.stderr}")
+
+
+def _rm_by_url(src_url: URL) -> None:
```

Comment:
minor: this method does only local file/dir removal, so it better to have the following signature:
```
def _rm_local(path: Path) -> None:
    if path.is_dir(): ...
```

Reply:
done in 
013ad1a